xla/autotuning.proto

```diff
@@ -38,6 +38,13 @@ message AutotuneResult {
     DISQUALIFIED = 3;
   }
 
+  enum Backend {
```

Comment:
But this is already encoded in the oneof key?

Reply:
The context of this development is that the XLA:GPU developers at Nvidia needs debugging information for a tuned fusion - the fusion name, each backend autotuner has tried (triton / cuDNN / cuBLAS) for the fusion and their run_time. As well as the best kernel with the fastest runtime. This helps us understand how each backend performs. 

We believe this would be helpful for upstreaming into XLA.

IIUC, AutotuneResults is a list that presents all tuned fusions. And each tuned fusion is presented by AutotuneResult. We believe this debugging information is tightly attached to each HLO fusion and should stick with AutotuneResult. 

WDYT?

Reply:
My point is that the AutotuneResults structure consists of a list of Entries (repeated Entry results) and each entry has one AutotuneResult currently. Instead of putting the results of multiple backends into the same AutotuneResult object, I think that we could add multiple AutotuneResult objects to the Entry (repeated AutotuneResult alternative_results), one for each backend. And keep "result" as the best result.

But I would to check with @cheshire also what does he recommend.

Reply:
Isn't this implicitly selected based on the type of the value of `oneof key`?

Reply:
Yeah each HLO/fusion has a selected best backend result implicitly encoded by oneof Key. We want to include other not-selected backend information for debugging purpose and hope to get the info dumped along with the autotune result.

Reply:
Right, I understand and agree with the rationale, I'm just saying that I think you're adding redundant data which is already there on top of the required change. Also : didn't you do stochastic rounding with us before ? :P

Reply:
Hehe hello George :) I'm making a mess somewhere else now! 
So IIUC, you're also suggesting we should make a repeated AutotuneResult list in the Entry of AutotuneResults? Maybe we should have all of your feedback earlier in the loop - next time :)

Reply:
Yes - let's try to go back to the drawing board on what are we aiming to achieve? The `AutotuningLog` already can have all the `AutotuningResult`s. Each autotuning result already know it's backend, based on the `key` value. AutotuningLog for each instruction already stored `instr` representation itself. Do we need any proto changes at all?

Reply:
(Hmm I didn't know about `AutotuningLog` when I wrote my comments, maybe that's more fitting than `AutotuneResults`.)